[Documentation] Add Mistralai docs (#623)
langchain4j/langchain4j: docs/docs/integrations/language-models/mistralai.md

Create a class and add the following code.

```java
import dev.langchain4j.data.message.AiMessage;
import dev.langchain4j.model.StreamingResponseHandler;
import dev.langchain4j.model.mistralai.MistralAiStreamingChatModel;
import dev.langchain4j.model.output.Response;

import java.util.concurrent.CompletableFuture;

public class HelloWorld {
    public static void main(String[] args) {
        MistralAiStreamingChatModel model = MistralAiStreamingChatModel
                .withApiKey(ApiKeys.MISTRALAI_API_KEY);

        CompletableFuture<Response<AiMessage>> futureResponse = new CompletableFuture<>();         
        model.generate("Tell me a joke about Java", new StreamingResponseHandler() {
            @Override
            public void onNext(String token) {
                System.out.print(token);
            }

            @Override
            public void onComplete(Response<AiMessage> response) {
                futureResponse.complete(response);
            }

            @Override
            public void onError(Throwable error) {
                futureResponse.completeExceptionally(error);
            }    
        });

        futureResponse.join();
    }
}
```

> You can receive each chunk text as it is generated by LLM on the `onNext` method.

Reviewer: ```suggestion
You will receive each chunk of text (token) as it is generated by the LLM on the `onNext` method.
```

You can see that output below is streamed in real-time.

```plaintext
"Why do Java developers wear glasses? Because they can't C#"
```

Of course, you can combine MistralAI chat completion with other features like [Set Model Parameters](/docs/tutorials/model-parameters) and [Chat Memory](/docs/tutorials/chat-memory) to get more accurate responses.

> In [Chat Memory](/docs/tutorials/chat-memory) you will learn how to pass along your chat history, so the LLM knows what has been said before. If you don't pass the chat history, like in this simple example, the LLM will not be able to correctly answer the second question ('What did I just ask?').

Reviewer: ```suggestion
In [Chat Memory](/docs/tutorials/chat-memory) you will learn how to pass along your chat history, so the LLM knows what has been said before. If you don't pass the chat history, like in this simple example, the LLM will not know what has been said before, so it won't be able to correctly answer the second question ('What did I just ask?').
```

A lot of parameters are set behind the scenes, such as timeout, model type and model parameters.
In [Set Model Parameters](/docs/tutorials/model-parameters) you will learn how to set these parameters explicitly.

### More examples

> if you want check more examples, you can find them in the [langchain4j-examples](https://github.com/langchain4j/langchain4j-examples) project.

Reviewer: ```suggestion
If you want to check more examples, you can find them in the [langchain4j-examples](https://github.com/langchain4j/langchain4j-examples) project.
All examples in the [langchain4j-examples/tutorials](https://github.com/langchain4j/langchain4j-examples/tutorials) folder can also be run with Mistral AI.
```

> if you want check more examples, you can find them in the [langchain4j-examples](https://github.com/langchain4j/langchain4j-examples) project.

Reviewer: Not sure if really all examples of the tutorial folder can be run with Mistral AI, feel free to remove that phrase again. In particular, I'm unsure if it has tools support? (if not, this would exclude the tools examples)

Author: It's true, I have removed the general reference to all the examples in `langchain4j-examples` for just the `mistralai-examples`. I think it'll be more accurate.
